android/src/main/java/com/tailscale/ipn/MainActivity.kt

```diff
@@ -64,62 +59,49 @@ class MainActivity : ComponentActivity() {
                             onNavigateToMDMSettings = { navController.navigate("mdmSettings") },
                             onNavigateToManagedBy = { navController.navigate("managedBy") })
 
-                    composable("main") {
-                        MainView(
-                            viewModel = MainViewModel(manager.model, manager),
-                            navigation = mainViewNav
+                    val exitNodePickerNav = ExitNodePickerNav(onNavigateHome = {
+                        navController.popBackStack(
```

Comment:
In my OG implementation of the exit node picker, I navigated back to home by using `navigage()`. This would push a new home page onto the stack without freeing the resources used by the old one. Using `popBackStack()` instead unwinds the stack and reuses the existing one, which also has the benefit of it loading faster.